src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -751,7 +751,19 @@ internal static bool IsPackageReferenceProject(Project project)
                         {
                             try
                             { // In case proj and assets file are not in sync and some refs were deleted
-                                installedPackage = projPackages.Where(p => p.Name.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
+                                var isPackageCentrallyManaged = tfmInformation.CentralPackageVersions.Any(cpv => cpv.Key.Equals(topLevelPackage.Name, StringComparison.Ordinal));
+                                if (isPackageCentrallyManaged)
```

Comment:
All packages are expected to be centrally managed, unless there's `VersionOverride`. Not sure if that check adds something.

Reply:
You're right, I can just check if the CPM is enabled. 

I did some testing and even if the user has `VersionOverride` it will appear as centrally managed since it's going to be declared in the Directory.Packages.props file too.